lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java

```diff
@@ -251,6 +253,7 @@ public class TrinoS3FileSystem
     private static final Set<String> GLACIER_STORAGE_CLASSES = ImmutableSet.of(Glacier.toString(), DeepArchive.toString());
     private static final MediaType DIRECTORY_MEDIA_TYPE = MediaType.create("application", "x-directory");
     private static final String S3_DEFAULT_ROLE_SESSION_NAME = "trino-session";
+    private static final String S3_ACCESS_DENIED_ERROR_CODE = "AccessDenied";
```

Comment:
nit: I would move it to before/after `NO_SUCH_*_ERROR_CODE` constants and rename to `ACCESS_DENIED_ERROR_CODE`.